network_analysis/find-connected-features-in-utility-network/src/main/java/com/esri/samples/find_connected_features_in_utility_network/FindConnectedFeaturesInUtilityNetworkController.java

```diff
@@ -0,0 +1,503 @@
+
+      // set whether the user is adding a starting point or a barrier
+      boolean isAddingStart = addingStartRadioButton.isSelected();
+
+      // get the clicked map point
+      Point2D screenPoint = new Point2D(e.getX(), e.getY());
+      Point mapPoint = mapView.screenToLocation(screenPoint);
+
+      // identify the feature to be used
+      ListenableFuture<List<IdentifyLayerResult>> identifyLayerResultsFuture = mapView.identifyLayersAsync(screenPoint, 10, false);
+      identifyLayerResultsFuture.addDoneListener(() -> {
+        try {
+          // get the result of the query
+          List<IdentifyLayerResult> identifyLayerResults = identifyLayerResultsFuture.get();
+
+          // return if no features are identified
+          if (identifyLayerResults.isEmpty()) {
+            return;
+
+          } else {
+            // retrieve the first result and get it's contents
+            IdentifyLayerResult firstResult = identifyLayerResults.get(0);
+            LayerContent layerContent = firstResult.getLayerContent();
+            // check that the result is a feature layer and has elements
+            if (layerContent instanceof FeatureLayer && !firstResult.getElements().isEmpty()) {
+              // retrieve the geoelements in the feature layer
+              GeoElement identifiedElement = firstResult.getElements().get(0);
+              if (identifiedElement instanceof ArcGISFeature) {
+                // get the feature
+                ArcGISFeature identifiedFeature = (ArcGISFeature) identifiedElement;
+
+                // get the network source of the identified feature
+                UtilityNetworkSource networkSource = utilityNetwork.getDefinition().getNetworkSource(identifiedFeature.getFeatureTable().getTableName());
+
+                UtilityElement utilityElement = null;
+
+                // check if the network source is a junction or an edge
+                if (networkSource.getSourceType() == UtilityNetworkSource.Type.JUNCTION) {
+                  //  create a utility element with the identified feature
+                  utilityElement = createUtilityElement(identifiedFeature, networkSource);
+                }
+                // check if the network source is an edge
+                else if (networkSource.getSourceType() == UtilityNetworkSource.Type.EDGE && identifiedFeature.getGeometry().getGeometryType() == GeometryType.POLYLINE) {
+
+                  //  create a utility element with the identified feature
+                  utilityElement = utilityNetwork.createElement(identifiedFeature, null);
+
+                  // get the geometry of the identified feature as a polyline, and remove the z component
+                  Polyline polyline = (Polyline) GeometryEngine.removeZ(identifiedFeature.getGeometry());
+
+                  // compute how far the clicked location is along the edge feature
+                  utilityElement.setFractionAlongEdge(GeometryEngine.fractionAlong(polyline, mapPoint, -1));
```

Comment:
I've noticed when clicking on add a starting location, and then clicking in a location (shown on the screen shot) it throws an ArcGISRuntimeException error:
`Invalid argument: Fraction along value must be in [0,1] range`. It looks like for some reason this location returns a NaN which I think is causing the error. @colinanderson I think this could this be a Java bug if we're allowing NaN to be passed in?

Reply:
![image](https://user-images.githubusercontent.com/36415565/68416190-e870a600-018b-11ea-8113-9685b6ad2686.png)
This is the location (it should be on the screen when loading the app the first time)

Reply:
Note: this happens at other joins too

Reply:
Thanks @Rachael-E , I've confirmed with @ZackAllen that this happens in .NET as well. We'd need to investigate?

Reply:
@colinanderson what do you think about this? Should we be raising an issue to investigate this? I'm unclear if it's a Java or a CAPI bug.

Reply:
While this is being fixed in core I've added an error message to inform the user that the barrier/starting point cant be created at that particular map point.